tests/unit/s2n_pq_kem_test.c

```diff
@@ -50,6 +54,19 @@ int main()
 {
     BEGIN_TEST();
 
+#if defined(OPENSSL_IS_AWSLC) && defined(AWS_LC_API_VERSION)
+    const unsigned long lc_vers = awslc_api_version_num();
+#else
+    const unsigned long lc_vers = SSLeay();
+#endif
+
+    /* If using non-FIPS AWS-LC >= v1.4.0 (API vers. 20), expect Kyber512 KEM from AWS-LC */
+    if (s2n_libcrypto_is_awslc() && lc_vers >= 20 && !s2n_libcrypto_is_fips()) {
+        EXPECT_TRUE(s2n_libcrypto_supports_kyber_512());
+    } else {
+        EXPECT_FALSE(s2n_libcrypto_supports_kyber_512());
```

Comment:
Is this EXPECT_FALSE check needed? If kyber is enabled when it shouldn't be, the build should fail, which means this check may not be needed in a test. I'm just worried that this test will start failing if kyber is added to AWSLC-FIPS. But maybe this isn't a concern!

Reply:
it's not strictly needed, i suppose, but I like to use tests to tightly define expected behavior at a given point in time. AWS-LC runs a CI check [building s2n and running its tests against AWS-LC PR changes](https://github.com/aws/aws-lc/blob/main/tests/ci/run_s2n_integration.sh#L61), so we'll catch this on the AWS-LC side if/when Kyber is enabled in AWS-LC-FIPS.